src/Database/Query.php

```diff
@@ -1105,73 +1105,6 @@ public function andWhere(ExpressionInterface|Closure|array|string $conditions, a
         return $this;
     }
 
-    /**
-     * Adds a single or multiple fields to be used in the ORDER clause for this query.
-     * Fields can be passed as an array of strings, array of expression
-     * objects, a single expression or a single string.
-     *
-     * If an array is passed, keys will be used as the field itself and the value will
-     * represent the order in which such field should be ordered. When called multiple
-     * times with the same fields as key, the last order definition will prevail over
-     * the others.
-     *
-     * By default this function will append any passed argument to the list of fields
-     * to be selected, unless the second argument is set to true.
-     *
-     * ### Examples:
-     *
-     * ```
-     * $query->orderBy(['title' => 'DESC', 'author_id' => 'ASC']);
-     * ```
-     *
-     * Produces:
-     *
-     * `ORDER BY title DESC, author_id ASC`
-     *
-     * ```
-     * $query
-     *     ->orderBy(['title' => $query->newExpr('DESC NULLS FIRST')])
-     *     ->orderBy('author_id');
-     * ```
-     *
-     * Will generate:
-     *
-     * `ORDER BY title DESC NULLS FIRST, author_id`
-     *
-     * ```
-     * $expression = $query->newExpr()->add(['id % 2 = 0']);
-     * $query->orderBy($expression)->orderBy(['title' => 'ASC']);
-     * ```
-     *
-     * and
-     *
-     * ```
-     * $query->orderBy(function ($exp, $query) {
-     *     return [$exp->add(['id % 2 = 0']), 'title' => 'ASC'];
-     * });
-     * ```
-     *
-     * Will both become:
-     *
-     * `ORDER BY (id %2 = 0), title ASC`
-     *
-     * Order fields/directions are not sanitized by the query builder.
-     * You should use an allowed list of fields/directions when passing
-     * in user-supplied data to `order()`.
-     *
-     * If you need to set complex expressions as order conditions, you
-     * should use `orderByAsc()` or `orderByDesc()`.
-     *
-     * @param \Cake\Database\ExpressionInterface|\Closure|array|string $fields fields to be added to the list
-     * @param bool $overwrite whether to reset order with field list or not
-     * @return $this
-     * @deprecated 5.0.0 Use orderBy() instead now that CollectionInterface methods are no longer proxied.
-     */
-    public function order(ExpressionInterface|Closure|array|string $fields, bool $overwrite = false)
```

Comment:
Should we make the query functions a 2 major deprecation?

Reply:
I don't think that's necessary. Most IDEs show a error/warning when using methods with the `@deprecated` annoation, so users would have already switched to the new methods.

Reply:
We can also start showing an explicit deprecation warning in the last 5.x minor before we release 6.0.